Server.Core/RequestBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿ using System;
+ using System.Text;
+
+namespace Server.Core
+{
+    public class RequestBuilder
+    {
+        private Request request;
+
+        public RequestBuilder()
+        {
+            request = new Request();
+        }
+
+        public Request BuildRequest()
+        {
+            return request;
+        }
+
+        public Request BuildRequestFromWholeMessage(byte[] requestMessage)
+        {
+            if (requestMessage.Length == 0)
+            {
+                SetRequestToNoValues();
+            }
+            else
+            {
+                ParseMessageBytes(requestMessage);
+            }
+            return request;
+        }
+
+        public void SetRequestHttpVersion(string httpVersion)
+        {
+            request.HttpVersion = httpVersion;
+        }
+
+        public void SetRequestUri(string uri)
+        {
+            request.Uri = uri;
+        }
+
+        public void SetRequestMethod(string method)
+        {
+            request.Method = method;
+        }
+
+        public void SetRequestHeaders(string[] headers)
```

Comment:
Is this specific enough? Would it be more valuable to have a `AddHeader` method which takes a key and value?